lib/.staging

```diff
@@ -51,30 +116,53 @@ function clone {
+		delete_temp_staging 5 "Unable to drop staging database tables.";
+	}
+	wp db export $STAGING_DIR/.export-sql --add-drop-table --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to export database.";
+	}
+	sed -i -e "s/$DB_PREFIX/staging_$DB_PREFIX/g" $STAGING_DIR/.export-sql || {
+		delete_temp_staging 5 "Unable to update database prefix.";
+	}
+	cd $STAGING_DIR || {
+		delete_temp_staging 5 "Unable to move to staging directory.";
+	}
+	wp db import $STAGING_DIR/.export-sql --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to import database.";
+	}
 	if [ "0" != "$USER_ID" ]
 		then
 		wp user meta update $USER_ID session_tokens "$SESSIONS" --format=json --skip-themes --skip-plugins --quiet
 	fi
-	wp option update staging_environment staging --skip-themes --skip-plugins --quiet || error 'Unable to set environment.'
+	wp option update staging_environment staging --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to set environment.";
+	}
 	move_files $PRODUCTION_DIR $STAGING_DIR
 	if [ -L $PRODUCTION_DIR/index.php ]
 		then
 			echo 'path='$STAGING_DIR > /nfssys/etc/wp_symink_watch/$(whoami).notify
 		else
 			WP_VER=$(wp core version)
-			wp core download --version=$WP_VER --force || error 'Unable to install WordPress in staging directory.'
+			wp core download --version=$WP_VER --force || {
+				delete_temp_staging 5 "Unable to install WordPress in staging directory.";
+			}
 	fi
-	wp search-replace $PRODUCTION_URL $STAGING_URL --skip-themes --skip-plugins --quiet || error 'Unable to update URL on staging.'
-	wp option update staging_config "$CONFIG" --format=json --path=$STAGING_DIR --skip-themes --skip-plugins --quiet || error 'Unable to import global config on staging.'
-	wp option update nfd_coming_soon 'true' --path=$STAGING_DIR --skip-themes --skip-plugins --quiet || error 'Unable to turn on Coming Soon page in staging.'
-	wp rewrite flush --path=$STAGING_DIR --skip-themes --skip-plugins --quiet || error 'Unable to flush rewrite rules.'
+	wp search-replace $PRODUCTION_URL $STAGING_URL --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to update URLs on staging.";
+	}
+	wp option update staging_config "$CONFIG" --format=json --path=$STAGING_DIR --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to import global config on staging.";
+	}
+	wp option update nfd_coming_soon 'true' --path=$STAGING_DIR --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to turn on Coming Soon page in staging.";
+	}
+	wp rewrite flush --path=$STAGING_DIR --skip-themes --skip-plugins --quiet || {
+		delete_temp_staging 5 "Unable to flush rewrite rules.";
+	}
 	rm $STAGING_DIR/.export-sql --force
 	rewrite_htaccess $STAGING_DIR
+	wp transient delete nfd_staging_lock --path="$PRODUCTION_DIR" --skip-themes --skip-plugins --quiet
```

Comment:
Is it necessary to run `wp transient delete nfd_staging_lock --path="$PRODUCTION_DIR" --skip-themes --skip-plugins --quiet` in every function? I see that we were already doing it at the end of the script, and we can also add it to the `error` function. This way, all cases would be covered and we can remove it from everywhere else?